src/log.h

```diff
@@ -185,35 +190,120 @@ class LogOutputBuffer : public ICombinedLogOutput {
 	Logger &m_logger;
 };
 
+#ifdef __ANDROID__
+class AndroidLogOutput : public ICombinedLogOutput {
+	public:
+		void logRaw(LogLevel lev, const std::string &line);
+};
+#endif
 
-extern StreamLogOutput stdout_output;
-extern StreamLogOutput stderr_output;
-extern std::ostream null_stream;
+class LogTarget {
+public:
+	// Must be thread-safe. These can be called from any thread.
+	virtual bool hasOutput() = 0;
+	virtual void log(const std::string &buf) = 0;
+};
 
-extern std::ostream *dout_con_ptr;
-extern std::ostream *derr_con_ptr;
-extern std::ostream *derr_server_ptr;
 
-extern Logger g_logger;
+class StreamProxy {
```

Comment:
It's mostly obvious but it'd be nice to have a short comment on these classes that describe their function.